sonar-plugin-api/src/main/java/org/sonar/api/server/ws/Response.java

```diff
@@ -41,11 +41,14 @@
      */
     Stream setStatus(int httpStatus);
 
+    /**
+     * Response stream. Beware that proper error recovery is not possible.
```

Comment:
I suggest to complete with "Beware that proper error recovery is not possible, {@link #setStatus(int)} must not be called after writing the first byte. A 500 HTTP error is returned in that case."